### ISSUE TITLE
Unable to load DLL 'python37'...

### ISSUE THREAD

**Issue by 0xzz** [henon/Python.Included#2]
I installed Python.Included with Nuget, but a simple test script prompts "Unable to load DLL 'python37'". I used Python.NET directly before, and, as I understand it, that error message is because the python.exe executable is not on PATH. (Python.NET seems only able to use the system distribution.) With Python.NET, a workaround is to modify PATH during run time. Is this also required by Python.Included?

Basically, the following doesn't run...

```
using Python.Runtime;
...
...
using (Py.GIL())
{
}
...
```

For some reason, if I also install Numpy.NET, and call a method from Numpy.NET first, and everything is fine.

```
using Numpy;
using Python.Runtime;
...
...
np.arange(1);
using (Py.GIL())
{
}
...
```

**Comment by henon** [comment]
Actually Python.Included packages minimal embedded Python and is able to deploy it in your home directory (if not yet done so) and does not care if there is a local Python installation or not. Numpy.NET uses that feature and initializes Python.Included correctly. So in order for your code to work you need to call the installer's SetupPython function. This is how Numpy.dll sets up Python.Included and installs its own packaged numpy wheel afterwards:

```C#
        private static PyObject InstallAndImport(bool force = false)
        {
            var installer = new Installer();
            installer.SetupPython(force).Wait();
            installer.InstallWheel(typeof(NumPy).Assembly, "numpy-1.16.3-cp37-cp37m-win_amd64.whl").Wait();
            PythonEngine.Initialize();
            var mod = Py.Import("numpy");
            return mod;
        }
```

**Comment by 0xzz** [comment]
Problem solved. Thank you!

**Comment by acrigney** [comment]
But installer is a static class so I just used
private static PyObject InstallAndImport(bool force = false)
        {
            //var installer = new Installer();
            Installer.SetupPython(force).Wait();
            //Installer.InstallWheel(typeof(NumPy).Assembly, "numpy-1.16.3-cp37-cp37m-win_amd64.whl").Wait();
            PythonEngine.Initialize();
            var mod = Py.Import("numpy");
            return mod;
        }
which works

**Comment by acrigney** [comment]
I was just just using Python.Included not Numpy.NET

**Comment by henon** [comment]
Right, `Installer` was changed to a static class in the meantime. I am glad your problem is solved.
